File: Scripts/results/generated_code_results/RN_gpt/Productivity/CourseTaskManagerApp/CalendarScreen.js
import React from 'react';
import { View, Text, StyleSheet, StatusBar, Image, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';

const CalendarScreen = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <StatusBar barStyle="dark-content" />
      <View style={styles.header}>
        <Text style={styles.dateText}>17 September</Text>
        <Text style={styles.todayText}>Today</Text>
        <TouchableOpacity
          style={styles.addButton}
          onPress={() => navigation.navigate('AddTask')}
        >
          <Text style={styles.addButtonText}>+ Add Task</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.calendarHeader}>
        <Text style={styles.calendarDay}>Mon</Text>
        <Text style={styles.calendarDay}>Tue</Text>
        <Text style={styles.calendarDay}>Wed</Text>
        <Text style={styles.highLightDay}>Thu 17</Text>
        <Text style={styles.calendarDay}>Fri</Text>
        <Text style={styles.calendarDay}>Sat</Text>
        <Text style={styles.calendarDay}>Sun</Text>
      </View>
      <View style={styles.scheduleList}>
        <View style={styles.scheduleItem}>
          <Text style={styles.timeText}>9:30</Text>
          <View style={styles.classDetail}>
            <Text style={styles.classTitle}>Physics</Text>
            <Text style={styles.classChapter}>Chapter 3: Force</Text>
            <Text style={styles.classInstructor}>Alex Jesus</Text>
            <Text style={styles.classPlatform}>Google Meet</Text>
          </View>
        </View>
        <View style={styles.scheduleItem}>
          <Text style={styles.timeText}>11:00</Text>
          <View style={styles.classDetail}>
            <Text style={styles.classTitle}>Geography</Text>
            <Text style={styles.classChapter}>Chapter 12: Soil Profile</Text>
            <Text style={styles.classInstructor}>Jenifer Clark</Text>
            <Text style={styles.classPlatform}>Zoom</Text>
          </View>
        </View>
        <View style={styles.scheduleItem}>
          <Text style={styles.timeText}>12:20</Text>
          <View style={styles.classDetail}>
            <Text style={styles.classTitle}>Assignment</Text>
            <Text style={styles.classChapter}>World Regional Pattern</Text>
            <Text style={styles.classInstructor}>Alexa Tenorio</Text>
            <Text style={styles.classPlatform}>Google Docs</Text>
          </View>
        </View>
      </View>
      <View style={styles.bottomNav}>
        <TouchableOpacity onPress={() => navigation.navigate('Home')}>
          <Icon name="home-outline" size={30} color="#666" />
        </TouchableOpacity>
        <TouchableOpacity onPress={() => navigation.navigate('Calendar')}>
          <Icon name="calendar-outline" size={30} color="#34a0a4" />
        </TouchableOpacity>
        <TouchableOpacity onPress={() => navigation.navigate('Classroom')}>
          <Icon name="school-outline" size={30} color="#666" />
        </TouchableOpacity>
        <TouchableOpacity onPress={() => navigation.navigate('Discussion')}>
          <Icon name="chat-outline" size={30} color="#666" />
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    paddingHorizontal: 20,
    paddingTop: 20,
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginBottom: 30,
  },
  dateText: {
    fontSize: 24,
    fontWeight: '600',
  },
  todayText: {
    fontSize: 18,
    marginLeft: 10,
    color: '#888',
  },
  addButton: {
    backgroundColor: '#34a0a4',
    padding: 10,
    borderRadius: 5,
  },
  addButtonText: {
    color: '#fff',
    fontSize: 16,
  },
  calendarHeader: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 20,
  },
  calendarDay: {
    fontSize: 16,
    color: '#888',
  },
  highLightDay: {
    fontSize: 16,
    color: '#34a0a4',
    fontWeight: '700',
  },
  scheduleList: {
    flex: 1,
  },
  scheduleItem: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 20,
  },
  timeText: {
    fontSize: 18,
    width: 60,
    color: '#555',
  },
  classDetail: {
    padding: 10,
    borderRadius: 10,
    backgroundColor: '#f8f9fa',
    flex: 1,
  },
  classTitle: {
    fontSize: 18,
    fontWeight: '600',
  },
  classChapter: {
    fontSize: 16,
    color: '#666',
  },
  classInstructor: {
    fontSize: 16,
    color: '#666',
  },
  classPlatform: {
    fontSize: 16,
    color: '#666',
  },
  bottomNav: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    paddingVertical: 10,
    borderTopWidth: 1,
    borderTopColor: '#ddd',
  },
});

export default CalendarScreen;

File: Scripts/results/generated_code_results/Flutter/Productivity/CourseTaskManagerApp/calendarScreen.dart
import 'package:flutter/material.dart';
class CalendarScreen extends StatefulWidget {
@override
_CalendarScreenState createState() => _CalendarScreenState();
}
class _CalendarScreenState extends State<CalendarScreen> {
final List<String> weekDays = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
final List<int> dates = [14, 15, 16, 17, 18, 19, 20];
int selectedDate = 17;
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
title: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text('17 September', style: TextStyle(fontSize: 16)),
Text('Today', style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold)),
],
),
actions: [
ElevatedButton.icon(
onPressed: () {
Navigator.pushNamed(context, 'AddTask');
},
icon: Icon(Icons.add),
label: Text('Add Task'),
style: ElevatedButton.styleFrom(
backgroundColor: Colors.green,
),
),
],
),
body: Column(
children: [
Container(
height: 80,
child: ListView.builder(
scrollDirection: Axis.horizontal,
itemCount: 7,
itemBuilder: (context, index) {
return GestureDetector(
onTap: () {
setState(() {
selectedDate = dates[index];
});
},
child: Container(
width: 50,
margin: EdgeInsets.symmetric(horizontal: 5),
decoration: BoxDecoration(
color: selectedDate == dates[index] ? Colors.blue : Colors.transparent,
borderRadius: BorderRadius.circular(10),
),
child: Column(
mainAxisAlignment: MainAxisAlignment.center,
children: [
Text(weekDays[index], style: TextStyle(color: selectedDate == dates[index] ? Colors.white : Colors.black)),
Text(dates[index].toString(), style: TextStyle(color: selectedDate == dates[index] ? Colors.white : Colors.black, fontWeight: FontWeight.bold)),
],
),
),
);
},
),
),
Expanded(
child: ListView(
children: [
_buildEventCard('9:30', '10:20', 'Physics', 'Chapter 3: Force', 'Alex Jesus', 'Google Meet', Colors.purple),
_buildEventCard('11:00', '11:30', 'Geography', 'Chapter 12: Soil Profile', 'Jennifer Clark', 'Zoom', Colors.green),
_buildEventCard('12:20', '13:00', 'Assignment', 'World Regional Pattern', 'Alexia Tenorio', 'Google Docs', Colors.orange),
],
),
),
],
),
bottomNavigationBar: BottomNavigationBar(
currentIndex: 1,
onTap: (index) {
if (index == 0) {
Navigator.pushNamed(context, 'Home');
}
},
items: [
BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Home'),
BottomNavigationBarItem(icon: Icon(Icons.list), label: 'Tasks'),
BottomNavigationBarItem(icon: Icon(Icons.school), label: 'Classroom'),
BottomNavigationBarItem(icon: Icon(Icons.chat), label: 'Chat'),
],
),
);
}
Widget _buildEventCard(String startTime, String endTime, String subject, String chapter, String instructor, String platform, Color color) {
return Card(
margin: EdgeInsets.symmetric(vertical: 8, horizontal: 16),
color: color,
child: Padding(
padding: EdgeInsets.all(16),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Row(
mainAxisAlignment: MainAxisAlignment.spaceBetween,
children: [
Text('$startTime - $endTime', style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold)),
Icon(Icons.more_vert, color: Colors.white),
],
),
SizedBox(height: 8),
Text(subject, style: TextStyle(color: Colors.white, fontSize: 18, fontWeight: FontWeight.bold)),
Text(chapter, style: TextStyle(color: Colors.white70)),
SizedBox(height: 8),
Row(
children: [
CircleAvatar(
backgroundImage: AssetImage('assets/images/template.png'),
radius: 12,
),
SizedBox(width: 8),
Text(instructor, style: TextStyle(color: Colors.white)),
Spacer(),
Icon(Icons.video_call, color: Colors.white, size: 20),
SizedBox(width: 4),
Text(platform, style: TextStyle(color: Colors.white)),
],
),
],
),
),
);
}
}